packages/kbn-zod-helpers/src/is_valid_date_math.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as z from 'zod';
+import { parseScheduleDates } from '@kbn/securitysolution-io-ts-utils';
+
+export function isValidDateMath(input: string, ctx: z.RefinementCtx) {
+  const parsedInput = parseScheduleDates(input);
```

Comment:
It's not great that it depends on `parseScheduleDates` from `@kbn/securitysolution-io-ts-utils`, because it creates a dependency of a generic package like `zod-helpers` onto a domain-specific package like `securitysolution-io-ts-utils` which should never be the case.

Also, `parseScheduleDates` depends on `moment` which we shouldn't need here for checking if it's a valid datemath expression.

Is there already a function for that in `@kbn/datemath` that we could use here?

Reply:
Agree 👍 I'll take a look if we can get rid of the `@kbn/securitysolution-io-ts-utils` dependency

Reply:
Done